pkg/kubelet/cm/devicemanager/device_plugin_stub.go

```diff
@@ -96,10 +103,17 @@ func (m *Stub) Start() error {
 	return nil
 }
 
-// Stop stops the gRPC server
+// Stop stops the gRPC server. Can be called without a prior Start
+// and more than once. Not safe to be called concurrently by different
+// goroutines!
 func (m *Stub) Stop() error {
+	if m.server == nil {
+		return nil
+	}
 	m.server.Stop()
-	close(m.stop)
+	m.wg.Wait()
+	m.server = nil
+	close(m.stop) // This prevents re-starting the server.
```

Comment:
You still need to close the stop chan even if the server is nil

Reply:
Fixed in the current revision.

Closing a closed channel panics, so now Stop() sets m.stop to nil and checks for that, to keep Stop() idempotent.

Reply:
Hmm, that turned out to be more complicated than anticipated. Setting it to nil was flagged as a race condition. I had to introduced a separate boolean stopIsOpen to track whether m.stop still needs to be closed, because there is no "Closed()" method.

Reply:
Actually, sorry for making you close that channel.
Turns out you don't need to do it as go channels are garbage collected and closing the channel is a control signal on the channel indicating that no more data follows.

Do you mind reverting that change?

Reply:
I had this garbage collection thing in mind tbh :)